packages/react-components/src/localization/LocalizationProvider.tsx

```diff
@@ -76,10 +76,74 @@ export type LocalizationProviderProps = {
  * @remarks Components will be provided localized strings in English (US) by default if this
  * provider is not used
  */
-export const LocalizationProvider = (props: LocalizationProviderProps): JSX.Element => {
+const InternalLocalizationProvider = (props: InternalLocalizationProviderProps): JSX.Element => {
   const { locale, children } = props;
   return <LocaleContext.Provider value={locale}>{children}</LocaleContext.Provider>;
 };
 
 /** React hook to access locale */
 export const useLocale = (): ComponentLocale => useContext(LocaleContext);
+
+interface SetLocaleContext {
+  setLocale: (locale?: string) => Promise<void>;
+}
+
+const SetLocaleContext = createContext<SetLocaleContext>({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
```

Comment:
Please drop `@typescript-esling/no-empty-function` by adding a comment in the function body about why this is empty. See https://eslint.org/docs/rules/no-empty-function

You can likely drop the `no-unsed-var` as well by not including that argument:

    const setLocaleContext = createContext<SetLocaleContext>({
        setLocale: async () => { /* this is empty because ... */ }
    });